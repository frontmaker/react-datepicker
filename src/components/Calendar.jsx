// @flow
import React from 'react';
import moment from 'moment';

import Weeks from './Weeks';

// import '../style/main.scss';

type Props = {
    date: Object,
}

type State = {
    date: Object,
    locale: string,
    selectedDate: Object
}

export default class Calendar extends React.Component {
  state: State;

  static defaultProps = {
    locale: 'en',
    date: moment(Date.now()),
  };

  prevMonth: Function;
  nextMonth: Function;
  selectDate: Function;

  constructor(props: Props) {
    super(props);

    this.state = {
      date: props.date,
			selectedDate: props.date.locale(props.locale),
      locale: props.locale,
    };

    this.prevMonth = this.prevMonth.bind(this);
    this.nextMonth = this.nextMonth.bind(this);
    this.selectDate = this.selectDate.bind(this);
  }

  prevMonth() {
    const { date } = this.props;
    date.add(-1, 'M');
    this.setState({ date });
  }

  nextMonth(): void {
    const { date } = this.props;
    date.add(1, 'M');
    this.setState({ date });
  }

  selectDate(date: Object): void {
    this.setState({ selectedDate: date });
  }

  render() {

    return (
      <div className="calendar-container">
        <div className="calendar-header">
          <div className="calendar-header__year">
            {this.state.selectedDate.format('YYYY')}
          </div>
          <div className="calendar-header__date">
            <span>{this.state.selectedDate.format('ddd')},</span>
            <span>{this.state.selectedDate.format('MMM')}</span>
            <span>{this.state.selectedDate.format('D')}</span>
          </div>
        </div>
        <div className="calendar-body">
          <div className="calendar-body__month">
            <button className="calendar-body__icon _left" onClick={this.prevMonth}>
              <i className="fa fa-angle-left">&nbsp;</i>
            </button>
            {this.state.selectedDate.format('MMMM')}
            <button className="calendar-body__icon _right" onClick={this.nextMonth}>
              <i className="fa fa-angle-right">&nbsp;</i>
            </button>
          </div>
          {this.renderWeekDays()}
          <Weeks
            select={this.selectDate}
            selectedDate={this.state.selectedDate}
            date={this.state.date}
          />
        </div>
      </div>
    );
  }

  renderWeekDays() {
		const currentDate = moment.locale(this.state.locale);

  	const weekDays = moment.weekdaysShort();

    return (
      <div className="calendar-weeks">
				{weekDays.map((day, index) => <span key={index} className="calendar-weeks__day">{day}</span>)}
      </div>
    );
  }
}
