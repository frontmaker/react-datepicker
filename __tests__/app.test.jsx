import React from 'react';
import renderer from 'react-test-renderer';
import Calendar from '../src/components/Calendar';


describe('Calendar', () => {
    it('render component without crashing', () => {
        const component = renderer.create(<Calendar />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    })
})