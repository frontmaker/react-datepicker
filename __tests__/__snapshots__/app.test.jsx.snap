exports[`Calendar render component without crashing 1`] = `
<div
  className="calendar-container">
  <div
    className="calendar-header">
    <div
      className="calendar-header__year">
      2017
    </div>
    <div
      className="calendar-header__date">
      <span>
        Tue
        ,
      </span>
      <span>
        Mar
      </span>
      <span>
        21
      </span>
    </div>
  </div>
  <div
    className="calendar-body">
    <div
      className="calendar-body__month">
      <button
        className="calendar-body__icon _left"
        onClick={[Function]}>
        <i
          className="fa fa-angle-left">
           
        </i>
      </button>
      March
      <button
        className="calendar-body__icon _right"
        onClick={[Function]}>
        <i
          className="fa fa-angle-right">
           
        </i>
      </button>
    </div>
    <div
      className="calendar-weeks">
      <span
        className="calendar-weeks__day">
        Mon
      </span>
      <span
        className="calendar-weeks__day">
        Tue
      </span>
      <span
        className="calendar-weeks__day">
        Wed
      </span>
      <span
        className="calendar-weeks__day">
        Thu
      </span>
      <span
        className="calendar-weeks__day">
        Fri
      </span>
      <span
        className="calendar-weeks__day">
        Sat
      </span>
      <span
        className="calendar-weeks__day">
        Sun
      </span>
    </div>
    <div>
      <div
        className="calendar-days">
        <div>
          <div
            className="calendar-days__empty" />
        </div>
        <div>
          <div
            className="calendar-days__empty" />
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              1
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              2
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              3
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              4
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              5
            </span>
          </button>
        </div>
      </div>
      <div
        className="calendar-days">
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              6
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              7
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              8
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              9
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              10
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              11
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              12
            </span>
          </button>
        </div>
      </div>
      <div
        className="calendar-days">
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              13
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              14
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              15
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              16
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              17
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              18
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              19
            </span>
          </button>
        </div>
      </div>
      <div
        className="calendar-days">
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              20
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day _selected _today"
            onClick={[Function]}>
            <span>
              21
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              22
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              23
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              24
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              25
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              26
            </span>
          </button>
        </div>
      </div>
      <div
        className="calendar-days">
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              27
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              28
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              29
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              30
            </span>
          </button>
        </div>
        <div>
          <button
            className="calendar-days__day"
            onClick={[Function]}>
            <span>
              31
            </span>
          </button>
        </div>
        <div>
          <div
            className="calendar-days__empty" />
        </div>
        <div>
          <div
            className="calendar-days__empty" />
        </div>
      </div>
    </div>
  </div>
</div>
`;
